src/store/product/reducer.js

```diff
@@ -3,12 +3,17 @@ const initialState = [];
 export default function reducer(state = initialState, action) {
   switch (action.type) {
     case "ALL_PRODUCTS": {
-      console.log("products in reducer?", action);
       return action.products;
     }
-    // case "ADD_PRODUCT": {
-    //   return [...state, action.newProduct];
-    // }
+    case "ADD_PRODUCT": {
+      return [...state, action.newProduct];
+    }
+    case "ONE_PRODUCT": {
+      return action.product;
```

Comment:
this seems weird, on one case you are returning an array as the state and in the other you return one product, this means the state of your reducers varies from array to object I assume. This is not good practice, think about turning it into an object like 
`const initialState = { list: [], single: null }` and you can call it `products` in the reducer so you can use it like `products.list or products.single` from your components